core/database_api.php

```diff
@@ -222,78 +226,75 @@ function db_check_database_support( $p_db_type ) {
 }
 
 /**
- * Checks if the database driver is MySQL
- * @return boolean true if mysql
+ * Maps a db driver type to the functional databse type
+ * @param string	$p_driver_type Database driver name
+ * @return int		Database type
  */
-function db_is_mysql() {
-	$t_db_type = config_get_global( 'db_type' );
-
-	switch( $t_db_type ) {
+function db_get_type( $p_driver_type ) {
+	switch( $p_driver_type ) {
 		case 'mysql':
 		case 'mysqli':
-			return true;
+			return DB_TYPE_MYSQL;
+		case 'postgres':
+		case 'postgres7':
+		case 'pgsql':
+			return DB_TYPE_PGSQL;
+		case 'mssql':
+		case 'mssqlnative':
+		case 'odbc_mssql':
+			return DB_TYPE_MSSQL;
+		case 'db2':
+			return DB_TYPE_DB2;
+		case 'oci8':
+			return DB_TYPE_ORACLE;
+		default:
+			return DB_TYPE_UNDEFINED;
```

Comment:
This is a quite unlikely scenario, but maybe we could trigger an error in case of unsupported DB ?